luigi/configuration/core.py

```diff
@@ -66,9 +66,26 @@ def add_config_path(path):
     # select parser by file extension
     _base, ext = os.path.splitext(path)
     if ext and ext[1:] in PARSERS:
-        parser_class = PARSERS[ext[1:]]
+        parser = ext[1:]
     else:
-        parser_class = PARSERS[PARSER]
+        parser = PARSER
+    parser_class = PARSERS[parser]
+
+    if not parser_class.enabled:
+        msg = (
+                "Parser not installed yet. "
+                "Please, install luigi with required parser:\n"
+                "pip install luigi[{parser}]"
+            )
+        raise ImportError(msg.format(parser=parser))
+
+    if parser != PARSER:
+        msg = (
+            "Config for {added} parser added, but used {used} parser. "
+            "Set up right parser via env var: "
+            "export LUIGI_CONFIG_PARSER={added}"
+        )
+        warnings.warn(msg.format(added=parser, used=PARSER))
```

Comment:
Does `warnings.warn()` also write to logs?

Reply:
I use it because `warnings.warn` used [in many other places in project](https://github.com/spotify/luigi/search?q=warnings.warn&unscoped_q=warnings.warn). Yes, [warnings can be captured](https://docs.python.org/3/library/logging.html#logging.captureWarnings).